src/+datools/+statistical/+ensemble/SIR.m

```diff
@@ -2,22 +2,25 @@
 
     methods
 
-        function analysis(obj, R, y)
+        function analysis(obj, obs)
 
             tau = obj.Rejuvenation;
-            tc = obj.Model.TimeSpan(1);
 
             xf = obj.Ensemble;
             xa = xf;
 
             ensN = obj.NumEnsemble;
             wf = obj.Weights;
 
-            Hxf = obj.Observation.observeWithoutError(tc, xf);
-            t0 = Hxf - y;
+            Hxf = obj.Observation.observeWithoutError(xf);
+            %t0 = Hxf - y;
 
-            dR = decomposition(R, 'chol');
-            as = exp(-0.5*sum(t0.*(dR \ t0), 1)).';
+            %dR = decomposition(R, 'chol');
+            %as = exp(-0.5*sum(t0.*(dR \ t0), 1)).';
+
+            as = obs.log(Hxf).';
+            m = max(as);
+            as = exp(as-(m + log(sum(exp(as-m)))));
```

Comment:
Can you elaborate a little more why we doing this?

Reply:
numerical stability.